src/main.cpp

```diff
@@ -220,15 +220,14 @@ int main(int argc, char **argv) {
             msg.wrench.torque.z = (data[5]-8192)/calib[5];
 
             pub.publish(msg);
+            lock.unlock();
         } else {
             // Request for offset reset
             write(fdc, "O", 1);
             offset_reset_ --;
+            lock.unlock();
             cv_.notify_all();
         }
-        lock.unlock();
-        ros::spinOnce();
```

Comment:
Bugfix: spinOnce should not be used here. The service callbacks are already handled by the AsyncSpinner threads. In fact, when this spinOnce handles a service callback itself (which happens from time to time), we get a deadlock